lib/srv/db/watcher_test.go

```diff
@@ -111,103 +87,156 @@ func TestWatcher(t *testing.T) {
-		sort.Sort(d)
-		require.Empty(t, cmp.Diff(types.Databases{db0, db1, db2}, d,
-			cmpopts.IgnoreFields(types.Metadata{}, "ID"),
-		))
-	case <-time.After(time.Second):
-		t.Fatal("Didn't receive reconcile event after 1s.")
-	}
+	assertReconciledResource(t, reconcileCh, types.Databases{db0, db1, db2})
 
 	// Update db1 labels so it doesn't match.
 	db1.SetStaticLabels(map[string]string{"group": "c", types.OriginLabel: types.OriginDynamic})
 	err = testCtx.authServer.UpdateDatabase(ctx, db1)
 	require.NoError(t, err)
 
 	// Only db0 and db2 should remain registered.
-	select {
-	case d := <-reconcileCh:
-		sort.Sort(d)
-		require.Empty(t, cmp.Diff(types.Databases{db0, db2}, d,
-			cmpopts.IgnoreFields(types.Metadata{}, "ID"),
-		))
-	case <-time.After(time.Second):
-		t.Fatal("Didn't receive reconcile event after 1s.")
-	}
+
+	assertReconciledResource(t, reconcileCh, types.Databases{db0, db2})
 
 	// Remove db2.
 	err = testCtx.authServer.DeleteDatabase(ctx, db2.GetName())
 	require.NoError(t, err)
 
 	// Only static database should remain.
+	assertReconciledResource(t, reconcileCh, types.Databases{db0})
+}
+
+// TestWatcherRDSDynamicResource RDS dynamic resource registration where the ResourceMatchers should be always
+// evaluated for the dynamic registered resources.
+func TestWatcherCloudDynamicResource(t *testing.T) {
+	ctx := context.Background()
+	testCtx := setupTestContext(ctx, t)
+
+	db0, err := makeStaticDatabase("db0", nil)
+	require.NoError(t, err)
+	reconcileCh := make(chan types.Databases)
+	testCtx.setupDatabaseServer(ctx, t, agentParams{
+		Databases: []types.Database{db0},
+		ResourceMatchers: []services.ResourceMatcher{
+			{Labels: types.Labels{
+				"group": []string{"a"},
+			}},
+		},
+		OnReconcile: func(d types.Databases) {
+			reconcileCh <- d
+		},
+	})
+	assertReconciledResource(t, reconcileCh, types.Databases{db0})
+
+	withRDSURL := func(v3 *types.DatabaseSpecV3) {
+		v3.URI = "mypostgresql.c6c8mwvfdgv0.us-west-2.rds.amazonaws.com"
+	}
+
+	// Created an RDS db dynamic resource that doesn't match any db service ResourceMatchers.
+	db1, err := makeDynamicDatabase("db1", map[string]string{"group": "z"}, withRDSURL)
+	require.NoError(t, err)
+	require.True(t, db1.IsRDS())
+	err = testCtx.authServer.CreateDatabase(ctx, db1)
+	require.NoError(t, err)
+	// The db1 should not be registered by the agent due to ResourceMatchers mismatch:
+	assertReconciledResource(t, reconcileCh, types.Databases{db0})
+
+	// Create an RDS dynamic resource with labels that matches ResourceMatchers.
+	db2, err := makeDynamicDatabase("db2", map[string]string{"group": "a"}, withRDSURL)
+	require.NoError(t, err)
+	require.True(t, db2.IsRDS())
+
+	err = testCtx.authServer.CreateDatabase(ctx, db2)
+	require.NoError(t, err)
+	// The db2 service should be properly registered by the agent.
+	assertReconciledResource(t, reconcileCh, types.Databases{db0, db2})
+
+	// Create an RDS Could resource with a label that doesn't match resource matcher.
```

Comment:
```suggestion
	// Create an RDS cloud resource with a label that doesn't match resource matcher.
```
Same typo for line #175, and `makeCouldDatabase`